--- a/include/vterm.h	2026-07-29 11:37:42.640290865 +0300
+++ b/include/vterm.h	2026-07-29 11:38:25.031902650 +0300
@@ -245,6 +245,7 @@
   VTERM_ATTR_BACKGROUND, // color:  40-49 100-107
   VTERM_ATTR_SMALL,      // bool:   73, 74, 75
   VTERM_ATTR_BASELINE,   // number: 73, 74, 75
+  VTERM_ATTR_DIM,        // bool:   2, 22
 
   VTERM_N_ATTRS
 } VTermAttr;
@@ -509,6 +510,7 @@
     unsigned int dhl       : 2; /* On a DECDHL line (1=top 2=bottom) */
     unsigned int small     : 1;
     unsigned int baseline  : 2;
+    unsigned int dim       : 1;
 } VTermScreenCellAttrs;
 
 enum {
@@ -589,8 +591,9 @@
   VTERM_ATTR_CONCEAL_MASK    = 1 << 9,
   VTERM_ATTR_SMALL_MASK      = 1 << 10,
   VTERM_ATTR_BASELINE_MASK   = 1 << 11,
+  VTERM_ATTR_DIM_MASK        = 1 << 12,
 
-  VTERM_ALL_ATTRS_MASK = (1 << 12) - 1
+  VTERM_ALL_ATTRS_MASK = (1 << 13) - 1
 } VTermAttrMask;
 
 int vterm_screen_get_attrs_extent(const VTermScreen *screen, VTermRect *extent, VTermPos pos, VTermAttrMask attrs);
--- a/src/vterm_internal.h	2026-07-29 11:37:42.641290856 +0300
+++ b/src/vterm_internal.h	2026-07-29 11:38:34.359817372 +0300
@@ -50,6 +50,7 @@
   unsigned int font:4; /* To store 0-9 */
   unsigned int small:1;
   unsigned int baseline:2;
+  unsigned int dim:1;
 };
 
 struct VTermState
--- a/src/pen.c	2026-07-29 11:37:42.641290856 +0300
+++ b/src/pen.c	2026-07-29 11:41:12.084382363 +0300
@@ -179,6 +179,7 @@
   state->pen.font = 0;      setpenattr_int (state, VTERM_ATTR_FONT, 0);
   state->pen.small = 0;     setpenattr_bool(state, VTERM_ATTR_SMALL, 0);
   state->pen.baseline = 0;  setpenattr_int (state, VTERM_ATTR_BASELINE, 0);
+  state->pen.dim = 0;       setpenattr_bool(state, VTERM_ATTR_DIM, 0);
 
   state->pen.fg = state->default_fg;  setpenattr_col(state, VTERM_ATTR_FOREGROUND, state->default_fg);
   state->pen.bg = state->default_bg;  setpenattr_col(state, VTERM_ATTR_BACKGROUND, state->default_bg);
@@ -202,6 +203,7 @@
     setpenattr_int (state, VTERM_ATTR_FONT,      state->pen.font);
     setpenattr_bool(state, VTERM_ATTR_SMALL,     state->pen.small);
     setpenattr_int (state, VTERM_ATTR_BASELINE,  state->pen.baseline);
+    setpenattr_bool(state, VTERM_ATTR_DIM,       state->pen.dim);
 
     setpenattr_col( state, VTERM_ATTR_FOREGROUND, state->pen.fg);
     setpenattr_col( state, VTERM_ATTR_BACKGROUND, state->pen.bg);
@@ -301,6 +303,11 @@
       break;
     }
 
+    case 2: // Faint (dim) on
+      state->pen.dim = 1;
+      setpenattr_bool(state, VTERM_ATTR_DIM, 1);
+      break;
+
     case 3: // Italic on
       state->pen.italic = 1;
       setpenattr_bool(state, VTERM_ATTR_ITALIC, 1);
@@ -359,9 +366,11 @@
       setpenattr_int(state, VTERM_ATTR_UNDERLINE, state->pen.underline);
       break;
 
-    case 22: // Bold off
+    case 22: // Normal intensity (bold and faint off)
       state->pen.bold = 0;
       setpenattr_bool(state, VTERM_ATTR_BOLD, 0);
+      state->pen.dim = 0;
+      setpenattr_bool(state, VTERM_ATTR_DIM, 0);
       break;
 
     case 23: // Italic and Gothic (currently unsupported) off
@@ -508,6 +517,9 @@
   if(state->pen.bold)
     args[argi++] = 1;
 
+  if(state->pen.dim)
+    args[argi++] = 2;
+
   if(state->pen.italic)
     args[argi++] = 3;
 
@@ -599,6 +611,10 @@
     val->number = state->pen.baseline;
     return 1;
 
+  case VTERM_ATTR_DIM:
+    val->boolean = state->pen.dim;
+    return 1;
+
   case VTERM_N_ATTRS:
     return 0;
   }
--- a/src/vterm.c	2026-07-29 11:37:42.641290856 +0300
+++ b/src/vterm.c	2026-07-29 11:39:41.134208324 +0300
@@ -278,6 +278,7 @@
     case VTERM_ATTR_BACKGROUND: return VTERM_VALUETYPE_COLOR;
     case VTERM_ATTR_SMALL:      return VTERM_VALUETYPE_BOOL;
     case VTERM_ATTR_BASELINE:   return VTERM_VALUETYPE_INT;
+    case VTERM_ATTR_DIM:        return VTERM_VALUETYPE_BOOL;
 
     case VTERM_N_ATTRS: return 0;
   }
--- a/src/screen.c	2026-07-29 11:37:42.641290856 +0300
+++ b/src/screen.c	2026-07-29 11:40:49.546586686 +0300
@@ -27,6 +27,7 @@
   unsigned int font      : 4; /* 0 to 9 */
   unsigned int small     : 1;
   unsigned int baseline  : 2;
+  unsigned int dim       : 1;
 
   /* Extra state storage that isn't strictly pen-related */
   unsigned int protected_cell : 1;
@@ -444,6 +445,9 @@
   case VTERM_ATTR_BASELINE:
     screen->pen.baseline = val->number;
     return 1;
+  case VTERM_ATTR_DIM:
+    screen->pen.dim = val->boolean;
+    return 1;
 
   case VTERM_N_ATTRS:
     return 0;
@@ -701,6 +705,7 @@
         dst->pen.font      = src->attrs.font;
         dst->pen.small     = src->attrs.small;
         dst->pen.baseline  = src->attrs.baseline;
+        dst->pen.dim       = src->attrs.dim;
 
         dst->pen.fg = src->fg;
         dst->pen.bg = src->bg;
@@ -990,6 +995,7 @@
   cell->attrs.font      = intcell->pen.font;
   cell->attrs.small     = intcell->pen.small;
   cell->attrs.baseline  = intcell->pen.baseline;
+  cell->attrs.dim       = intcell->pen.dim;
 
   cell->attrs.dwl = intcell->pen.dwl;
   cell->attrs.dhl = intcell->pen.dhl;
@@ -1120,6 +1126,8 @@
     return 1;
   if((attrs & VTERM_ATTR_BASELINE_MASK)    && (a->pen.baseline != b->pen.baseline))
     return 1;
+  if((attrs & VTERM_ATTR_DIM_MASK)    && (a->pen.dim != b->pen.dim))
+    return 1;
 
   return 0;
 }
